src/Raven.Server/Web/System/DatabasesHandler.cs

```diff
@@ -89,67 +89,76 @@ public Task GetTopology()
             }
 
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
```

Comment:
@haludi can you point me to the change in this code and the reasoning behind it?

Reply:
In general, if we try to get the topology of deleted database we will get an error with `["Database-Missing"]` in the header.
There is a race condition when the database is in delete progress so we can get empty topology.
I changed it to return `["Database-Missing"]` in the header in this situation as well.
```
if (dbRecord.Topology.Members.Count == 0 && dbRecord.Topology.Rehabs.Count == 0 && dbRecord.DeletionInProgress.Any())
{
    // The database at deletion progress from all nodes
    HttpContext.Response.StatusCode = (int)HttpStatusCode.ServiceUnavailable;
    HttpContext.Response.Headers["Database-Missing"] = name;
    using var writer = new BlittableJsonTextWriter(context, HttpContext.Response.Body);
    context.Write(writer, new DynamicJsonValue
        {
            ["Type"] = "Error",
            ["Message"] = "Database " + name + " was deleted"
        });
}
```